File: compiler/bench/swap2.fs
\ expected: 500000000
\ Swap pairs, 500M times
: main
  0 1                       ( a b )
  500000000 0 do
    swap                    ( b a )
    swap                    ( a b ) back to original
    swap 1 + swap           ( a' b )
  loop
  drop . cr
;

File: compiler/bench/swap2.c
// expected: 500000000
// Swap pairs, 500M times
#include <stdio.h>

int main(void) {
    long a = 0, b = 1;
    long tmp;
    for (long i = 0; i < 500000000L; i++) {
        // swap
        tmp = a; a = b; b = tmp;
        // swap back
        tmp = a; a = b; b = tmp;
        a = a + 1;
    }
    (void)b;
    printf("%ld\n", a);
    return 0;
}
